Utilities/Mergers/src/MergerAlgorithm.cxx

```diff
@@ -128,6 +130,20 @@ void merge(TObject* const target, TObject* const other)
   }
 }
 
+void merge(VectorOfTObjectPtr& targets, const VectorOfTObjectPtr& others)
+{
+  for (const auto& other : others) {
+    if (const auto target_same_name = std::find_if(targets.begin(), targets.end(), [&other](const auto& target) {
+          return std::string_view{other->GetName()} == std::string_view{target->GetName()};
+        });
+        target_same_name != targets.end()) {
+      merge(target_same_name->get(), other.get());
+    } else {
+      targets.push_back(other);
```

Comment:
I am wondering if it wouldn't be better to be consistent with [merging TCollections](https://github.com/AliceO2Group/AliceO2/blob/e7d0c164ba90dd7bbfed4e5a9ca815c9ff7d8063/Utilities/Mergers/src/MergerAlgorithm.cxx#L80), i.e. if an object does not exist in the target, we clone it. It has an advantage that the `other` vector/collection is not modified, thus it is less misleading when testing and possibly in the other future calls in the source code.

Reply:
That is not that bad of an idea, However I am really not sure, how to do it properly, as Clone method on TObject requires new name and I am not sure which name should I give it

Reply:
I don't think ROOT is going to complain about duplicate names if you `Clone`, at least I did not see that due to the use of `Clone` when merging `TCollections`. Does it produce a warning in your case?